modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/SegmentsExperimentRelModelListenerTest.java

```diff
@@ -0,0 +1,259 @@
+						"{experimentId}",
+						String.valueOf(
+							segmentsExperiment.getSegmentsExperimentKey())),
+					() -> "{}")));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			_addSegmentsExperimentRel(segmentsExperience, segmentsExperiment);
+
+			Assert.assertNotNull(
+				_segmentsExperimentRelLocalService.getSegmentsExperimentRel(
+					segmentsExperiment.getSegmentsExperimentId(),
+					segmentsExperience.getSegmentsExperienceId()));
+		}
+		finally {
+			ReflectionTestUtil.setFieldValue(
+				asahFaroBackendClient, "_http", _http);
+		}
+	}
+
+	@Test
+	public void testOnAfterRemove() throws Exception {
+		SegmentsExperiment segmentsExperiment = _addSegmentsExperiment();
+
+		SegmentsExperience segmentsExperience = _addSegmentsExperience();
+
+		SegmentsExperimentRel segmentsExperimentRel = _addSegmentsExperimentRel(
+			segmentsExperience, segmentsExperiment);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					StringUtil.replace(
+						"/api/1.0/experiments/{experimentId}/dxp-variants",
+						"{experimentId}",
+						String.valueOf(
+							segmentsExperiment.getSegmentsExperimentKey())),
+					() -> "{}")));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			_segmentsExperimentRelLocalService.deleteSegmentsExperimentRel(
+				segmentsExperimentRel.getSegmentsExperimentRelId());
+
+			Assert.assertNull(
```

Comment:
Here you are not asserting the update call to AC, so maybe this test should be translated into a unit test or not tested at all.

Reply:
This class has been deleted and I added some unit tests of AsahFaroBackendClientImplTest class.